src/libsyntax/parse/mod.rs

```diff
@@ -466,6 +466,7 @@ crate fn lit_token(lit: token::Lit, suf: Option<Symbol>, diag: Option<(Span, &Ha
     match lit {
        token::Byte(i) => (true, Some(LitKind::Byte(byte_lit(&i.as_str()).0))),
        token::Char(i) => (true, Some(LitKind::Char(char_lit(&i.as_str(), diag).0))),
+       token::Err(i) => (true, Some(LitKind::Char(char_lit(&i.as_str(), diag).0))),
```

Comment:
I believe this has to be (and add a `LitKind::Err` variant to the enum)

```suggestion
       token::Err(i) => (true, Some(LitKind::Err)),
```
in order to get rid of the following error, but we can tackle that later:

```
error[E0277]: can't compare `&str` with `char`
  --> $DIR/lex-bad-char-literals-6.rs:8:10
   |
LL |     if x == z {}  // no error here
   |          ^^ no implementation for `&str == char`
   |
   = help: the trait `std::cmp::PartialEq<char>` is not implemented for `&str`
```

Changing this will have knock down effects that will need to be handled appropriately, eventually giving the resulting expression a type of `TyErr`.